lib/sycamore/sycamore/query/execution/sycamore_operator.py

```diff
@@ -473,21 +468,26 @@ def script(self, input_var: Optional[str] = None, output_var: Optional[str] = No
         if self.s3_cache_path:
             cache_string = f", cache=S3Cache('{self.s3_cache_path}')"
         result = f"""
-{output_var or get_var_name(logical_node)} = {input_var or get_var_name(logical_node.dependencies[0])}.map(
-    lambda doc: llm_extract_operation(
-        client=OpenAI(OpenAIModels.GPT_4O.value{cache_string}),
-        doc=doc,
-        question='{question}',
-        new_field='{new_field}',
-        field='{field}',
-        format='{fmt}',
-        discrete={discrete},
-    ),
-    **{self.get_node_args()},
-)
-"""
+        messages = EntityExtractorMessagesPrompt(
+                question='{question}', field='{field}', format='{fmt}, discrete={discrete}
+            ).get_messages_dict()
+
+        entity_extractor = OpenAIEntityExtractor(
```

Comment:
will likely simplify this code soon, share an entityextractor that just accepts new field names